assets/javascripts/discourse/lib/topic-level-assign-menu.js

```diff
@@ -0,0 +1,227 @@
+import { getOwner } from "@ember/application";
```

Comment:
Can you point me the new code here please?

Reply:
Yeah, sorry, it should be difficult to distinguish where is the old code and where the implementation of the new buttons. I extracted commits with refactoring and the new code. So now you can review it separately:
1. [REFACTOR: Extract topic level assign menu code into separate file](https://github.com/discourse/discourse-assign/pull/554/commits/1cd7ba75bf744fa64037efff9d8899ba8b6e5542)
2. [REFACTOR: Extract methods](https://github.com/discourse/discourse-assign/pull/554/commits/0933b743c54233240a7d5215bad7259759054fdc) (nothing new was added on this stage, I've just extracted some methods to make further changes easier)
3. [Implement](https://github.com/discourse/discourse-assign/pull/554/commits/8cfc9f2a3e6e528a16fb8096a1b3001f5653e5f5)